PayloadPkg/OsLoader/LoadImage.c

```diff
@@ -616,6 +616,7 @@ LoadBootImages (
     }
 
     LoadedImage = (LOADED_IMAGE *)AllocateZeroPool (sizeof (LOADED_IMAGE));
+    ASSERT (LoadedImage);

```

Comment:
I think better not to ASSERT.   
Since it is non-critical, error flow is preferred instead.